### PR TITLE
InukaPageView instrumentation

### PR DESCRIPTION
**Pull request by stephanebisson**
Heavily inspired by https://github.com/wikimedia/mediawiki-extensions-WikimediaEvents/blob/master/modules/ext.wikimediaEvents/InukaPageView.js

Schema: https://meta.wikimedia.org/w/index.php?title=Schema:InukaPageView&oldid=19739286
Phabricator Link: https://phabricator.wikimedia.org/T242358

Todo
- [x] Handle visibility change. Keep track of how long the page is hidden.
- [x] Keep track of how many sections are viewed
- [x] Confirm whether we need `beforeunload` event in the device << it doesn't fire window.beforeunload, so we use `pagehide`
- [x] `referring_domain` - `kaios-app` can be the value within app
- [x] Confirm the values chosen for `base url`, `webHost` and `wiki` with analytics people on phab
    - [x] `webHost`, either `en.wikipedia.org` or `en.m.wikipedia.org`
    - [x] `wiki`, [language]wiki `enwiki` `hiwiki`
- [x] <s>Adjust UA</s>, add `app_version` key
- [x] <s>(TBC) access_method `mobile` , https://phabricator.wikimedia.org/T246309</s>

https://phabricator.wikimedia.org/T242358

### CODE REVIEW

File: src/hooks/useTracking.js [hunk 1]
```diff
@@ -63,8 +63,14 @@ export const useTracking = (
...
   sectionCount = 0,
-  openedSections = 0
+  openedSections = {}
 ) => {
+  const trackingRef = useRef()
+
+  useEffect(() => {
+    trackingRef.openedSectionCount = Object.keys(openedSections).length
+  }, [openedSections])
+
```
Comment (hueitan) [inline src/hooks/useTracking.js]: Note :

We cannot simply use `openedSections` here, the reason is given here https://github.com/facebook/react/issues/14965

the suggestion is using `useRef` to handle things like mutable value
https://reactjs.org/docs/hooks-reference.html#useref

**Comment by hueitan** [comment]
NOTE: 
I don't find a way to send beacon when the app exit, and we also need to make sure the `back/exit` button works the same in different devices.

**Comment by jpita** [comment]
` @hueitan hueitan marked this pull request as ready for review 8 days ago` 

what's the status of this?
should I check it somehow?

File: src/hooks/useTracking.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import { useRef, useEffect } from 'preact/hooks'
+
+const SCHEMA_NAME = 'InukaPageView'
+const SCHEMA_REV = 19739286
+const USER_ID_KEY = 'INUKA-PV-U'
+const SESSION_ID_KEY = 'INUKA-PV-S'
+const ONE_HOUR = 3600 * 1000
+const BASE_URL = 'https://en.wikipedia.org/beacon/event'
```
Comment (stephanebisson) [inline src/hooks/useTracking.js]: Should this URL vary based on current language?

**Comment by stephanebisson** [comment]
> * [ ]  (TBC) access_method `mobile` , https://phabricator.wikimedia.org/T246309
This is for the VirtualPageView schema. Unrelated to this.

File: src/hooks/useTracking.js (hunk 2 above)
Comment (hueitan) [reply src/hooks/useTracking.js]: Event Logging uses `webHost` `wiki` to determine the article language, upon checking with team there is no need to change the base url.

**Comment by hueitan** [comment]
The code is updated with the latest Schema now.

File: src/hooks/useTracking.js [hunk 3]
```diff
@@ -103,6 +103,7 @@ export const useTracking = (
           session_id: getSessionId(),
           pageview_token: pageviewToken,
           client_type: 'kaios-app',
+          app_version: '1.0.0',
```
Comment (stephanebisson) [inline src/hooks/useTracking.js]: We should probably read this from package.json

Comment (hueitan) [reply src/hooks/useTracking.js]: Great! This is added here https://github.com/wikimedia/wikipedia-kaios/pull/131/commits/6be5f9a7dbcd634d3dd815123698da2d671a682f

**Comment by jpita** [comment]
is this sending metrics to somewhere I can check?

**Pull request merged**